### PR TITLE
User signin and signup

### PR DESCRIPTION
**Pull request by Adil-sc**
- Implemented React Context to be able to share "user" data across components
user is stored in [user, setUser] state in App.js
All routes are wrapped with <UserContext.Provider> and passed in [user, and setUser] from App.js so user can be set/accessed from other components
- Added functionality to login if user exists
- Added functionality to signup new user

### CODE REVIEW

File: Startup.cs [hunk 1]
```diff
@@ -34,6 +34,8 @@ public void ConfigureServices(IServiceCollection services)
         options.UseNpgsql(sqlConfig);
       });
 
+
```
Comment (buckeyez) [inline Startup.cs:37]: no changes to startup pls :)

File: Startup.cs [hunk 2]
```diff
@@ -34,6 +34,8 @@ public void ConfigureServices(IServiceCollection services)
         options.UseNpgsql(sqlConfig);
       });
 
+      
+
```
Comment (michellepana) [inline Startup.cs:38]: delete the two blank lines?

File: package.json [hunk 3]
```diff
@@ -21,5 +21,10 @@
   "homepage": "https://github.com/buckeyez/OSU-CS467-Software_Quiz#readme",
   "devDependencies": {
     "babel-plugin-macros": "^3.0.1"
+  },
+  "dependencies": {
```
Comment (buckeyez) [inline package.json:25]: all this should be in client app

File: ClientApp/src/utils/signupUser.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    password: password,
+    lastName: lastName,
+    firstName: firstName,
+  };
+  const headers = {
+    'Access-Control-Allow-Origin': 'https://localhost:5001/',
+  };
+
+  const response = await axios
+    .post('https://localhost:5001/users/add', data, headers)
```
Comment (michellepana) [inline ClientApp/src/utils/signupUser.js:26]: As per our discussion, we can update to relative path

**Review by buckeyez — APPROVED** [review]

**Review by buckeyez — APPROVED** [review]

**Pull request merged**